ecl/hql/hqlattr.cpp

```diff
@@ -3776,6 +3776,71 @@ ITypeInfo * setStreamedAttr(ITypeInfo * _type, bool setValue)
     }
 }
 
+//---------------------------------------------------------------------------------
+double queryLikelihood(IHqlExpression * expr)
+{
+    IInterface * match = meta.queryExistingProperty(expr, EPlikelihood);
+    if (match)
+    {
+        IHqlExpression * likelihoodExpr = static_cast<IHqlExpression *>(match);
+        IValue * value = likelihoodExpr->queryChild(0)->queryValue();
+        return value->getRealValue();
+    }
+
+    double likelihoodValue = -1.0;
+    switch(expr->getOperator())
+    {
+    case no_likely:
+        if (expr->numChildren() > 1)
+        {
+            IValue *value = expr->queryChild(1)->queryValue();
+            likelihoodValue = value->getRealValue();
+        }
+        else
+            likelihoodValue = 0.99;
```

Comment:
0.99 should be a static constant at the head of the file.  You should also consider creating a static IHqlExpression so this can become

`likelihoodExpr.set(likelyExpr);